### DISCUSSION TITLE
Tomcat instrumentation span names are not updatable from SpanProcessor

### DISCUSSION THREAD

**Discussion by srinusanduri** [open-telemetry/opentelemetry-java-instrumentation#7853] [Q&A] [answered]
I am trying to update span names from the SpanProcessor onStart method to redact sensitive path segments in HTTP request related span names. Span name update is working for most of auto instrumented spans except for `io.opentelemetry.tomcat-7.0` library auto instrumented spans. I verified for manual instrumentation spans, kafka, grcp, spring webmvc, http-url-connection auto instrumented spans for which span name update from SpanProcessor is working fine. Below is an example trace screenshot
![span_name_update](https://user-images.githubusercontent.com/10183113/219825422-51e0e25f-3fec-4c5d-ba39-6ea88a3bc7e0.png)

On further debugging, I observed that in `SpanProcessor.onStart` tomcat span name is received as `HTTP GET` which is then later updated to actual path `/student` before exporting somewhere after span start.

Also, I want to use tomcat span name for sampling decision. I want to drop health check requests (for example: `/actuator/health`), thought of using tomcat span name comparison to identify health check requests. But since tomcat span has name `HTTP GET` in the beginning of its lifecycle, this is not possible.

Why is tomcat span name is set as `HTTP GET` at start and being updated to actual path in the end? Which seems to be a deviation from all other library instrumentation spans. Its also not honoring the span name update from SpanProcessor.

**Comment by laurit** [comment, marked as answer]
See https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/ In tomcat instrumentation we are not able to determine a route so we start the span with a generic name. Later when request processing reaches a framework where we can get a route we'll update the span with a new better name. If you wish to prevent updating the span name you could try using https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b139e1d4295c9325e0af353afb884d36bec3197a/instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpRouteHolder.java#L58-L60 Set the source to `CONTROLLER` to prevent further updates.
As in sampler you can only use the information that is available when the span is started you are better off using `http.target` attribute instead of the span name.
